Doc/whatsnew/3.12.rst

```diff
@@ -171,6 +171,18 @@ APIs:
 * :func:`unittest.getTestCaseNames` (:gh:`50096`)
 * :class:`webbrowser.MacOSX` (:gh:`86421`)
 
+
+Pending Removal in Python 3.14
+==============================
+
+The following modules and APIs have been deprecated in earlier Python releases,
```

Comment:
Also mention this under `Deprecated` that it is deprecated in  3.12

Reply:
@kumaraditya303 it's not correct for this notice to go in `Pending Removal in Python 3.14`, it's supposed to be a reminder of previous deprecations, so I moved it to 'Deprecations'

Reply:
> oh this is the first deprecation of the 3.12 release cycle!

Nah, we already deprecated `typing.Sized` and `typing.Hashable` (in favour of `collections.abc.Sized` and `collections.abc.Hashable`) we just didn't set a removal date for those deprecations ;)

Reply:
bah you just got merged before me :p